cumulus/tasks/sf-sns-report/index.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const get = require('lodash.get');
+const { StepFunction } = require('@cumulus/ingest/aws');
+const { setGranuleStatus, sns } = require('@cumulus/common/aws');
+const errors = require('@cumulus/common/errors');
+const cumulusMessageAdapter = require('@cumulus/cumulus-message-adapter-js');
+
+/**
+ * Determines if there was a valid exception in the input message
+ *
+ * @param {Object} event - aws event object
+ * @returns {boolean} true if there was an exception, false otherwise
+ */
+function eventFailed(event) {
+  if (event.exception) {
+    if (typeof event.exception === 'object') {
+      // this is needed to avoid flagging cases like "exception: {}" or "exception: 'none'"
+      if (Object.keys(event.exception).length > 0) {
+        return true;
+      }
+    }
+  }
+  // Error and error keys are not part of the cumulus message
+  // and if they appear in the message something is seriously wrong
+  else if (event.Error || event.error) {
+    return true;
+  }
+  return false;
+}
+
+/**
+ * if the cumulus message shows that a previous step failed,
+ * this function extract the error message from the cumulus message
```

Comment:
`this function extracts the error message`

Reply:
done